the-collab-lab/tcl-26-smart-shopping-list
# Add item panel

## Description

This PR moves the AddItemForm from it's own page to the ListView page, in order to create a sliding panel on the same page as the shopping list. It also refactors the way the listItems data is shared between components, allowing us to reuse the data retrieved from Firestore for both the ShoppingList and the AddItemForm component (which previously fetched a copy of the list every time a user tried to add an item).

This was really tricky, particularly because I wanted to be sure larger font sizes wouldn't get cut off on the add form and the user would still be able to scroll. I ended up using a combination of fixed positioning for a background and absolute positioning in an area I knew would be empty, but allowing any amount of height for the form.

Possibly still some a11y issues at certain small sizes where it functions more like a modal, but not sure how to target JavaScript based on screen size effectively in the way you can CSS.

## Related Issue

Closes #47 
Closes #32 

Merged into this branch are several other related PRs, so also:
Closes #99 
Closes #98 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
| ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### After

![add panel](https://user-images.githubusercontent.com/67769490/132115305-3467a7c9-752f-4d35-af77-c1e321eb19eb.gif)

## Testing Steps / QA Criteria

1. Pull the branch, install dependencies and run npm start.
2. Test opening the add item form at different screen sizes.
3. Try adding some items to make sure it's working properly..
